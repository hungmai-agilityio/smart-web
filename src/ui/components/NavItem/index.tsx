'use client';
import Link from 'next/link';

interface NavItemProps {
  name: string;
  url?: string;
}

export const NavItem = ({ name, url }: NavItemProps) => {
  return (
    <Link href={url || '#'}>
      <p className="font-bold lg:text-lg md:text-base text-xs hover:text-second uppercase mb-3">
        {name}
      </p>
    </Link>
  );
};
