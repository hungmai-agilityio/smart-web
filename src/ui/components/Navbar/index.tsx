import { NavItem } from '@/ui/components/Navbar/Item';

interface NavbarProps {
  items: Array<{ name: string; url?: string }>;
  styles?: string;
}

export const Navbar = ({ items, styles }: NavbarProps) => {
  return (
    <nav className={styles}>
      {items.map((item, index) => (
        <NavItem key={index} name={item.name} url={item.url} />
      ))}
    </nav>
  );
};
