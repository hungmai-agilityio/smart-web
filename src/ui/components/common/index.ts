export * from '@/ui/components/common/Button';
export * from '@/ui/components/common/Dropdown';
export * from '@/ui/components/common/Icon';
export * from '@/ui/components/common/Photo';
export * from '@/ui/components/common/Text';
export * from '@/ui/components/common/StatsRow';
