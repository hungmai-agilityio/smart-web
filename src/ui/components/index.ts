export * from '@/ui/components/Hero';
export * from '@/ui/components/NavItem';
export * from '@/ui/components/Portfolio';
export * from '@/ui/components/Pagination';
export * from '@/ui/components/Skeleton/Card';
export * from '@/ui/components/Skeleton/Image';
export * from '@/ui/components/common';
