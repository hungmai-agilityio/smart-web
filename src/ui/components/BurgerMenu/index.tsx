'use client';

import Image from 'next/image';
import { useRef, useState } from 'react';
import { Navbar } from '@/ui/components';
import { useOutsideClick } from '@/hooks';

interface BurgerProps {
  items: Array<{ name: string; url?: string }>;
}

export const BurgerMenu = ({ items }: BurgerProps) => {
  const menuRef = useRef<HTMLDivElement>(null);
  const [menuOpen, setMenuOpen] = useState<boolean>(false);

  const toggleMenu = () => {
    setMenuOpen(!menuOpen);
  };

  useOutsideClick({
    ref: menuRef,
    handler: () => setMenuOpen(false)
  });

  return (
    <div ref={menuRef}>
      <div
        className="md:hidden flex items-center cursor-pointer"
        onClick={toggleMenu}
      >
        <Image src="/menu.svg" alt="menu" width={16} height={14} priority />
      </div>

      <Navbar
        items={items}
        styles={`md:flex gap-7 ${menuOpen ? 'block' : 'hidden'}
          md:relative absolute top-full left-0 w-full md:w-auto z-10 transition-all ease-in-out
          bg-white p-5 drop-shadow-lg md:drop-shadow-none lg:mr-28 md:mr-8`}
      />
    </div>
  );
};
