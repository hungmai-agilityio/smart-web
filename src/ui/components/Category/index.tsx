'use client';

import { memo, useCallback } from 'react';
import { useRouter, useSearchParams } from 'next/navigation';

// Constants
import { abel } from '@/constants';

// Components
import { Dropdown, ItemProps } from '@/ui/components';

interface CategoryProps {
  categories: string[];
  queryParam: string;
}

export const Category = memo(({ categories, queryParam }: CategoryProps) => {
  const router = useRouter();
  const searchParams = useSearchParams();

  const currentCategory = searchParams.get(queryParam);

  // Update category and URL when a new category is selected
  const handleCategorySelect = useCallback(
    (category: string) => {
      const params = new URLSearchParams(searchParams.toString());
      params.set(queryParam, category);
      router.push(`?${params.toString()}`, { scroll: false });
    },
    [searchParams, router, queryParam]
  );

  const dropdownItems: ItemProps[] = categories.map((category) => ({
    label: category,
    value: category
  }));

  return (
    <>
      <ul
        className={`${abel.className} my-8 md:flex hidden lg:gap-48 gap-16 justify-center`}
      >
        {categories.map((category, index) => (
          <li
            className={`uppercase cursor-pointer hover:text-second ${currentCategory === category ? 'text-second' : ''}`}
            key={index}
            onClick={handleCategorySelect.bind(null, category)}
          >
            {category}
          </li>
        ))}
      </ul>
      <div className="md:hidden flex justify-center my-8">
        <Dropdown
          items={dropdownItems}
          selectedValue={currentCategory!}
          onSelect={(value) => handleCategorySelect(value)}
        />
      </div>
    </>
  );
});
