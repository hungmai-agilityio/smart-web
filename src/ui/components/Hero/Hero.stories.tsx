import { Description, Heading } from '@/ui/components/common';
import { Hero } from '@/ui/components/Hero';
import { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Hero> = {
  component: Hero,
  tags: ['autodocs'],
  argTypes: {
    imageUrl: {
      description: 'URL of the image to be displayed'
    },
    children: {
      description: 'Content inside Hero that users can add'
    },
    buttonName: {
      description: 'Name the button'
    }
  }
};

export default meta;

type Story = StoryObj<typeof Hero>;

export const Default: Story = {
  args: {
    imageUrl: '/background-hero.webp'
  }
};

export const WithChildren: Story = {
  args: {
    imageUrl: '/background-hero.webp',
    children: (
      <>
        <Heading styles="text-white">Welcome to SmartWeb</Heading>
        <Description styles="font-thin text-white lg:text-3xl md:text-sm text-xs mt-14">
          Discover a World of Technological Advancements
        </Description>
      </>
    )
  }
};

export const WithButton: Story = {
  args: {
    imageUrl: '/background-hero.webp',
    buttonName: 'Button Name'
  }
};
