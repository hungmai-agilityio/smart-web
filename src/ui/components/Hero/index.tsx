import { ReactNode } from 'react';

// Constants
import { TYPE } from '@/constants';

// Components
import { Button, Photo } from '@/ui/components/common';

interface HeroProps {
  imageUrl: string;
  children?: ReactNode;
  buttonName?: string;
}

export const Hero = ({ imageUrl, children, buttonName }: HeroProps) => {
  return (
    <div className="relative w-full h-hero">
      <Photo src={imageUrl} alt="Hero Background" autoSize priority />
      <div className="absolute  inset-0 flex flex-col items-center justify-center">
        {children}
        {buttonName && (
          <div className="mt-10">
            <Button variant={TYPE.PRIMARY} name={buttonName} />
          </div>
        )}
      </div>
    </div>
  );
};
