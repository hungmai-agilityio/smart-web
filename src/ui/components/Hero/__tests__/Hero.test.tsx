import { Hero } from '@/ui/components/Hero';
import { render, screen } from '@testing-library/react';

describe('Hero component', () => {
  const defaultProps = {
    imageUrl: '/background-hero.webp',
    buttonName: 'Button Name',
    children: <p>Text content</p>
  };

  it('Should render default props', () => {
    render(<Hero {...defaultProps} />);
    expect(screen.getByAltText('Hero Background')).toBeInTheDocument();
    expect(screen.getByText('Text content')).toBeInTheDocument();
  });

  it('should be render match to snapshot', () => {
    const { container } = render(<Hero {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
