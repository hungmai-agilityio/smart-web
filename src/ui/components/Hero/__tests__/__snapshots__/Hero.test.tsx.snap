// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component should be render match to snapshot 1`] = `
<div>
  <div
    class="relative container w-full h-hero"
  >
    <div
      class="relative w-full h-full"
    >
      <img
        alt="Hero Background"
        class="transition-opacity duration-300"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        sizes="100vw"
        src="/_next/image?url=%2Fbackground-hero.webp&w=3840&q=75"
        srcset="/_next/image?url=%2Fbackground-hero.webp&w=640&q=75 640w, /_next/image?url=%2Fbackground-hero.webp&w=750&q=75 750w, /_next/image?url=%2Fbackground-hero.webp&w=828&q=75 828w, /_next/image?url=%2Fbackground-hero.webp&w=1080&q=75 1080w, /_next/image?url=%2Fbackground-hero.webp&w=1200&q=75 1200w, /_next/image?url=%2Fbackground-hero.webp&w=1920&q=75 1920w, /_next/image?url=%2Fbackground-hero.webp&w=2048&q=75 2048w, /_next/image?url=%2Fbackground-hero.webp&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </div>
    <div
      class="absolute  inset-0 flex flex-col items-center justify-center"
    >
      <p>
        Text content
      </p>
      <div
        class="mt-10"
      >
        <button
          class="text-primary px-2 rounded-full font-light truncate w-24 h-6 text-xs sm:w-28 sm:h-8 sm:text-sm md:w-36 md:h-10 md:text-base lg:w-52 lg:h-12 lg:text-lg bg-second"
        >
          Button Name
        </button>
      </div>
    </div>
  </div>
</div>
`;
